config/routes.rb

```diff
@@ -1,10 +1,14 @@
 Rails.application.routes.draw do
-    root "users#index"
-  
-    resources :users, only: [:index, :show] do
-      resources :posts, only: [:index, :new, :show, :create] do
-        resources :comments, only: [:new, :create]
-        resources :likes, only: [:create]
-      end
+  devise_for :users
+
+  devise_scope :user do
+    root to: "devise/sessions#new"
+  end
+
+  resources :users, only: [:index, :show] do
+    resources :posts, only: [:index, :new, :show, :create] do
+      resources :comments, only: [:new, :create]
+      resources :likes, only: [:create]
     end
-  end
\ No newline at end of file
+  end
```

Comment:
- [ ] [OPTIONAL] Please fix the routes to redirect the user to devise page only when the user is not authenticated. The current routing configuration is causing an infinite redirection loop, resulting in a page crash. To resolve this issue, ensure that the routes are configured to redirect unauthenticated users to the Devise login page and redirect authenticated users to the appropriate authenticated pages. So instead of this:

```rb
Rails.application.routes.draw do
  devise_for :users

  devise_scope :user do
    root to: "devise/sessions#new"
  end

  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :new, :show, :create] do
      resources :comments, only: [:new, :create]
      resources :likes, only: [:create]
    end
  end
end


```

Do this

```rb
Rails.application.routes.draw do
  devise_for :users

  authenticate :user do
    root to: "users#index", as: :authenticated_root # this is the path for authenticated users
  end

  unauthenticated :user do
    root to: "devise/sessions#new", as: :unauthenticated_root # this is the path for unauthenticated users
  end

  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :new, :show, :create] do
      resources :comments, only: [:new, :create]
      resources :likes, only: [:create]
    end
  end
end


```